### DISCUSSION TITLE
Hyperparameter tuning process

### DISCUSSION THREAD

**Discussion by avramit5674** [NVIDIA/NVFlare#2652] [Q&A]
### Python version (`python3 -V`)

3.5

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

2.3.8

### Operating system

Ubuntu 20.04

### Have you successfully run any of the following examples?

- [ ] hello-numpy-sag with simulator
- [ ] hello-pt with simulator
- [X] hello-numpy-sag with POC
- [ ] hello-pt with POC

### Please describe your question

I am currently setting up XGBoost for regression tasks using NVFlare, and I need to configure the hyperparameters in the server configuration files. I aim to tune these hyperparameters based on the data from all clients. For example, I plan to use Bayesian optimization for hyperparameter tuning, which requires access to the training data distributed across all clients. How can I effectively utilize optimization methods for hyperparameter tuning with NVFlare in a decentralized data environment?

**Comment by YuanTingHsieh** [comment]
I am not familiar with bayesian optimization for hyperparameter tuning.

When you said "access to the training data distributed across all clients.", what does this mean?
If server can access the full content of these distributed data, it defeats the purpose of federated learning
